src/client/LiveStream.js

```diff
@@ -515,10 +515,18 @@ exports.StreamStatus = async function({name, stopLro=false, showParams=false}) {
       return status;
     }
 
+    const segDurationMeta = await this.ContentObjectMetadata({
+      libraryId,
+      objectId,
+      metadataSubtree: "live_recording/recording_config/recording_params/xc_params/seg_duration"
+    });
+
+    const segDuration = segDurationMeta ? (parseInt(segDurationMeta) + 5) : 32.9;
```

Comment:
For safety let's change the logic so if seg_duration is not found in edgeMeta, we skip this test (so if state is running and seg_duration is defined and sinceLastFinalize > seg duration + 5

Reply:
If seg_duration is not found, the value used will default to 32.9, so `sinceLastFinalize > 32.9` when determining if the stream is stalled. Do you want to skip this logic entirely if seg_duration is not defined in the edgeMeta?

```
else if(state === "running" && sinceLastFinalize > segDuration) {
      state = "stalled";
    }
```

Reply:
Yes we need to skip the stalled logic if seg_duration is undefined because then our stalled logic would be bad (we would call it stalled when it isn't).

And add 5 sec allowance

So like this:

```
} else if(state === "running" && segDurationMeta && sinceLastFinalize > segDurationMeta + 5) {
      state = "stalled";
```